settings: Add function for saving (unchanged) single item/subtree

nordicjm opened the pull request:
Adds a function that allows saving of a single setting item without changing the value, or saving a subtree of settings, to the storage backend

nordicjm committed:
settings: Add function for saving (unchanged) single item/subtree

Adds a function that allows saving of a single setting item without
changing the value, or saving a subtree of settings, to the storage
backend

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

rmsaad commented on subsys/settings/src/settings_store.c:
```diff
@@ -303,3 +303,76 @@ void settings_store_init(void)
+#ifdef CONFIG_SETTINGS_SAVE_SINGLE_SUBTREE_WITHOUT_CHANGE
+int settings_save_subtree_or_single_without_modification(const char *name,
+							 bool save_if_subtree,
+							 bool save_if_single_setting);
```
the function definition shouldn’t have a semicolon at the end.

rmsaad commented:
@nordicjm These changes will work for my application, thanks.

nordicjm committed:
tests: settings: functional: Add test for save single function

Adds a test for the newly introduced
settings_save_subtree_or_single_without_modification() settings
function which allows saving a single value or subtree to NVM
without modifying the values

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: settings: functional: Enable single settings check

Enables the Kconfig for the newly introduced
settings_save_subtree_or_single_without_modification() settings
function so that it is tested in CI

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
mgmt: mcumgr: grp: settings_mgmt: Allow saving single setting

Adds functionality that allows saving a single setting to NVM using
the newly added function in the settings subsystem. This also
replaces calling the subtree save function if it the underlying
Kconfig is enabled to reduce code paths in settings mgmt

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
tests: mgmt: mcumgr: settings_mgmt: Add test for single setting save

Expands the test to check thr functionality of the newly adding save
subtree/save single setting

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
doc: releases: release-notes: 4.4: Add note on new settings Kconfigs

Lists new Kconfigs added to settings allowing single settings to be
saved to NVM without modifying their value

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

nordicjm committed:
doc: services: device_mgmt: smp_groups: Add note on single setting

Adds a note that saving single settings is supported, when the
required Kconfig is enabled

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>

fabiobaltieri reviewed (approved):
That's a very long option name, guess it's fine since it matches the function though.
